### PR TITLE
Update newVersion script to edit mobile-sdk-android files

### PR DESCRIPTION
**Pull request by toddsouthenbentley**

### CODE REVIEW

File: newVersion.py [hunk 1]
```diff
@@ -155,25 +163,44 @@ def modify_package_resolved(args, filename):
             package = match.group(1)
         if package == 'itwin-mobile-native':
             line = re.sub('("version": )".*"', '\\1"' + args.new_add_on + '"', line)
-            if (args.new_add_on_commit_id):
+            if hasattr(args, 'new_add_on_commit_id'):
```
Comment (tcobbs-bentley) [inline newVersion.py:166]: Should you also check that the value is non-null?

File: newVersion.py [hunk 2]
```diff
@@ -155,25 +163,44 @@ def modify_package_resolved(args, filename):
...
                 line = re.sub('("revision": )".*"', '\\1"' + args.new_add_on_commit_id + '"', line)
         elif package == 'itwin-mobile-sdk':
             line = re.sub('("version": )".*"', '\\1"' + args.new_mobile + '"', line)
-            if (args.new_commit_id):
+            if hasattr(args, 'new_commit_id'):
```
Comment (tcobbs-bentley) [inline newVersion.py:170]: Same here: non-null check?

File: newVersion.py [hunk 3]
```diff
@@ -155,25 +163,44 @@ def modify_package_resolved(args, filename):
...
+        ("( +ref: ')[.0-9a-z-]+", "\\g<1>" + args.new_add_on),
+        ("( +gh release download )[.0-9a-z-]+", "\\g<1>" + args.new_add_on),
+    ]) != 2:
+        raise Exception("Not enough replacements")
+
 def change_command(args):
     if not args.force:
         ensure_no_dirs_have_diffs()
-    if not args.current_mobile:
+    if not hasattr(args, 'current_mobile'):
```
Comment (tcobbs-bentley) [inline newVersion.py:193]: non-null check?

File: newVersion.py (hunk 1 above)
Comment (toddsouthenbentley) [reply newVersion.py:166]: I'm not sure, possibly. How would it get a null value if it has the attribute?

**Review by tcobbs-bentley — COMMENTED** [review]
I finished reviewing, leaving some comments.

File: newVersion.py (hunk 1 above)
Comment (tcobbs-bentley) [reply newVersion.py:166]: I'm worried that if the old code worked before your changes, that your new code won't work. I ran into that problem when I made a similar change. I had to update it to look like this:

    if not hasattr(args, 'new_mobile') or not args.new_mobile:

Comment (toddsouthenbentley) [reply newVersion.py]: Fixed

File: newVersion.py (hunk 2 above)
Comment (toddsouthenbentley) [reply newVersion.py]: Fixed

File: newVersion.py (hunk 3 above)
Comment (toddsouthenbentley) [reply newVersion.py]: Fixed

**Review by tcobbs-bentley — APPROVED** [review]

**Pull request merged**